lib/wlib/strings/DynamicString.h

```diff
@@ -0,0 +1,476 @@
+ * @param rhs @code DynamicString string as right hand side string
+ * @return true or false based on if two strings are equal
+ */
+
+bool operator== (const DynamicString &lhs, const DynamicString &rhs){
+	return lhs.compare(rhs) == 0;
+}
+
+/**
+ * Comparative operator compares if two strings are equal or not
+ *
+ * @param lhs character string as left hand side string
+ * @param rhs @code DynamicString string as right hand side string
+ * @return true or false based on if two strings are equal
+ */
+bool operator== (const char *lhs, const DynamicString &rhs){
+	return rhs.compare(lhs) == 0;
+}
+
+/**
+ * Comparative operator compares if two strings are equal or not
+ *
+ * @param lhs @code DynamicString string as left hand side string
+ * @param rhs character string as right hand side string
+ * @return true or false based on if two strings are equal
+ */
+bool operator== (const DynamicString &lhs, const char *rhs){
+	return lhs.compare(rhs) == 0;
+}
+
+/**
+ * Comparative operator compares if string and single character are equal or not
+ *
+ * @param lhs character as left hand side
+ * @param rhs @code DynamicString string as right hand side string
+ * @return true or false based on if two strings are equal
+ */
+bool operator== (const char lhs, const DynamicString &rhs){
+	return rhs.compare(lhs) == 0;
+}
+
+/**
+ * Comparative operator compares if string and single character are equal or not
+ *
+ * @param lhs @code DynamicString string as left hand side string
+ * @param rhs character as right hand side
+ * @return true or false based on if two strings are equal
+ */
+bool operator== (const DynamicString &lhs, const char rhs){
+	return lhs.compare(rhs) == 0;
+}
+
+/**
+ * Additive operator adds two given strings
+ *
+ * @param lhs @code DynamicString string as left hand side string
+ * @param rhs @code DynamicString string as right hand side string
+ * @return a @code DynamicString that is build from left hand string and right hand string
+ */
+DynamicString operator+(const DynamicString &lhs, const DynamicString &rhs){
+	DynamicString newStr;
+	newStr.append(lhs).append(rhs);
+	return newStr;
+}
+
+/**
+ * Additive operator adds two given strings
+ *
+ * @param lhs character string as left hand side string
+ * @param rhs @code DynamicString string as right hand side string
+ * @return a @code DynamicString that is build from left hand string and right hand string
+ */
+DynamicString operator+(const char *lhs, const DynamicString &rhs){
+	DynamicString newStr;
+	newStr.append(lhs).append(rhs);
+	return newStr;
+}
+
+/**
+ * Additive operator adds two given strings
+ *
+ * @param lhs @code DynamicString string as left hand side string
+ * @param rhs character string as right hand side string
+ * @return a @code DynamicString that is build from left hand string and right hand string
+ */
+DynamicString operator+(const DynamicString &lhs, const char *rhs){
+	DynamicString newStr;
+	newStr.append(lhs).append(rhs);

```

Comment:
Doing it this way would require two allocations and two copy operations. I'd rewrite this code (and all the other code like it) to require only a single allocation and copy of each `lhs` and `rhs`

Reply:
So no new string construction, and just append the rhs to lhs?

Reply:
You can create a constructor that takes two strings, allocates an array of size lhs.length + rhs.length + 1 (for null char) and then copies the contents of the two strings in

Reply:
Addition operator should not assign to or modify the arguments